File: app/src/main/java/me/aluceps/sandbox/MyApplication.java
package me.aluceps.sandbox;

import android.app.Application;
import android.support.annotation.NonNull;

import me.aluceps.sandbox.di.AppComponent;
import me.aluceps.sandbox.di.AppModule;
import me.aluceps.sandbox.di.DaggerAppComponent;
import timber.log.Timber;

public class MyApplication extends Application {

    AppComponent appComponent;

    @NonNull
    public AppComponent getComponent() {
        return appComponent;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        initializeInjector();
        initializeTimber();
    }

    private void initializeInjector() {
        appComponent = DaggerAppComponent.builder()
                .appModule(new AppModule(this))
                .build();
        appComponent.inject(this);
    }

    private void initializeTimber() {
        if (BuildConfig.DEBUG) {
            Timber.plant(new Timber.DebugTree());
        }
    }
}

File: app/src/main/java/me/aluceps/sandbox/MyApplication.kt
package me.aluceps.sandbox

import android.app.Application

import com.facebook.stetho.Stetho

import me.aluceps.sandbox.di.AppComponent
import me.aluceps.sandbox.di.AppModule
import me.aluceps.sandbox.di.DaggerAppComponent
import timber.log.Timber

class MyApplication : Application() {

    lateinit var component: AppComponent

    override fun onCreate() {
        super.onCreate()
        initializeInjector()
        initializeDebugUtilities()
    }

    private fun initializeInjector() {
        component = DaggerAppComponent.builder()
                .appModule(AppModule(this))
                .build()
        component.inject(this)
    }

    private fun initializeDebugUtilities() {
        if (BuildConfig.DEBUG) {
            Timber.plant(Timber.DebugTree())
            Stetho.initializeWithDefaults(this)
        }
    }
}
